drivers/sensor/bmm150/bmm150.c

```diff
@@ -0,0 +1,602 @@
+/* bmm150.c - Driver for Bosch BMM150 Geomagnetic Sensor */
+
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bmm150.h"
+
+static const struct {
+	int freq;
+	u8_t reg_val;
+} bmm150_samp_freq_table[] = { {2, 0x01},
+			       {6, 0x02},
+			       {8, 0x03},
+			       {10, 0x00},
+			       {15, 0x04},
+			       {20, 0x05},
+			       {25, 0x06},
+			       {30, 0x07} 
+};
+
+static const struct bmm150_preset {
+	u8_t rep_xy;
+	u8_t rep_z;
+	u8_t odr;
+} bmm150_presets_table[] = {
+	[LOW_POWER_PRESET] = {3 , 3, 10},
+	[REGULAR_PRESET] = {9, 15, 10},
+	[ENHANCED_REGULAR_PRESET] = {15, 27, 10},
+	[HIGH_ACCURACY_PRESET] = {47, 83, 20}
+};
+
+static int bmm150_set_power_mode(struct device *dev,
+				enum bmm150_power_modes mode,
+				int state)
+{
+	struct bmm150_data *data = dev->driver_data;
+	const struct bmm150_config *config = dev->config->config_info;
+
+	switch(mode){
+		case BMM150_POWER_MODE_SUSPEND:
+			if(i2c_reg_update_byte(data->i2c,
+						config->i2c_slave_addr,
+						BMM150_REG_POWER,
+						BMM150_MASK_POWER_CTL,
+						!state) < 0) {
+				return -EIO;
+			}
+			k_busy_wait(5 * USEC_PER_MSEC);
+			
+			return 0;
+		case BMM150_POWER_MODE_SLEEP:
+			if (i2c_reg_update_byte(data->i2c,
+						   config->i2c_slave_addr,
+						   BMM150_REG_OPMODE_ODR,
+						   BMM150_MASK_OPMODE,
+						   BMM150_MODE_SLEEP <<
+						   BMM150_SHIFT_OPMODE) < 0) {
+				return -EIO;
+			}
+			return 0;
```

Comment:
Break a line, I am picky about that but lines breaks are cheap and is makes the code breathing a bit, more readable imo. (rule being: break a line after a '}' unless it's another '}' or an else etc..). Apply that everywhere.